x-pack/plugins/reporting/server/export_types/csv_v2/csv_v2.ts

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import Boom from '@hapi/boom';
+import { KibanaRequest, IClusterClient } from '@kbn/core/server';
+import { DiscoverServerPluginStart } from '@kbn/discover-plugin/server';
+import { DataPluginStart } from '@kbn/data-plugin/server/plugin';
+import { CsvGenerator } from '@kbn/generate-csv';
+import { Writable } from 'stream';
+import { CancellationToken } from '@kbn/reporting-common';
+import { CONTENT_TYPE_CSV } from '@kbn/generate-csv/src/constants';
+import { JobParamsCsvFromSavedObject, TaskPayloadCsvFromSavedObject } from '../../../common/types';
+import {
+  CSV_REPORT_TYPE_V2,
+  LICENSE_TYPE_CLOUD_STANDARD,
+  LICENSE_TYPE_ENTERPRISE,
+  LICENSE_TYPE_GOLD,
+  LICENSE_TYPE_PLATINUM,
+  LICENSE_TYPE_TRIAL,
+} from '../../../common/constants';
+import { ExportType, ExportTypeSetupDeps, ExportTypeStartDeps } from '../common';
+import { ReportingRequestHandlerContext } from '../../types';
+import { getFieldFormats } from '../../services';
+import { decryptJobHeaders } from '../common/decrypt_job_headers';
+
+/*
+ * @TODO move to be within @kbn/reporitng-export-types
+ */
+
+export interface CsvExportTypeStartDeps extends ExportTypeStartDeps {
+  discover: DiscoverServerPluginStart;
+  data: DataPluginStart;
+  esClient: IClusterClient;
+}
+
+export class CsvV2ExportType extends ExportType<
+  JobParamsCsvFromSavedObject,
+  TaskPayloadCsvFromSavedObject
+> {
+  id = CSV_REPORT_TYPE_V2;
+  name = 'csv_v2';
+  jobType = CONTENT_TYPE_CSV;
+  jobContentEncoding = 'base64' as const;
+  jobContentExtension = 'csv' as const;
+  validLicenses = [
+    LICENSE_TYPE_TRIAL,
+    LICENSE_TYPE_CLOUD_STANDARD,
+    LICENSE_TYPE_GOLD,
+    LICENSE_TYPE_PLATINUM,
+    LICENSE_TYPE_ENTERPRISE,
+  ];
+  declare startDeps: CsvExportTypeStartDeps;
+
+  constructor(...args: ConstructorParameters<typeof ExportType>) {
+    super(...args);
+    const logger = args[2];
+    this.logger = logger.get('csv-export-v2');
+  }
+
+  setup(setupDeps: ExportTypeSetupDeps) {
+    this.setupDeps = setupDeps;
+  }
+
+  start(startDeps: CsvExportTypeStartDeps) {
+    this.startDeps = startDeps;
+  }
+
+  public getEsClient() {
+    return this.startDeps.esClient;
+  }
+
+  public createJob = async (
```

Comment:
This is failing the typescript currently, because the CreateJobFn type is not allowed to return a promise. The declarations for `RunTaskFn` and `CreateJobFn` should be reverted back to the version from `main`

Reply:
Leaving this comment for future me if I refer back to this:
From PR https://github.com/rshen91/kibana/pull/6
```
export type CreateJobFn<JobParamsType = BaseParams, JobPayloadType = BasePayload> = (
  jobParams: JobParamsType,
  context: ReportingRequestHandlerContext,
  req: KibanaRequest
) => Promise<Omit<JobPayloadType, 'headers' | 'spaceId'>>;
```

🙇🏻